# Copyright 2026 The OpenXLA Authors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ==============================================================================
# Triton compiles their CUDA launcher at runtime in order to not need CUDA
# just to build the Triton library and compile modules. We need to compile it
# ahead of time in order to run the tests correctly.

--- a/third_party/nvidia/backend/driver.py
+++ b/third_party/nvidia/backend/driver.py
@@ -98,13 +98,8 @@ class CudaUtils(object):
         return cls.instance
 
     def __init__(self):
-        mod = compile_module_from_file(
-            src_path=os.path.join(dirname, "driver.c"),
-            name="cuda_utils",
-            library_dirs=library_dirs(),
-            include_dirs=include_dirs,
-            libraries=libraries,
-        )
+        from ._C import cuda_utils
+        mod = cuda_utils
         global PyCUtensorMap
         global PyKernelArg
         global ARG_CONSTEXPR

